app/components/EventDate.tsx

```diff
@@ -1,24 +1,34 @@
-const DATE_OPTIONS: Intl.DateTimeFormatOptions = {
+const DATE_ONLY_OPTIONS: Intl.DateTimeFormatOptions = {
   weekday: "short",
   month: "short",
   day: "numeric",
-  hour: "numeric", // "7"
-  minute: "2-digit", // "00"
-  hour12: true, // "PM"
   timeZone: "UTC"
 };
 
-// TODO: make two date_options, one for EventDate, another for EventDateTime
+const DATE_TIME_OPTIONS: Intl.DateTimeFormatOptions = {
+  weekday: "short",
+  month: "short",
+  day: "numeric",
+  hour: "numeric",
+  minute: "2-digit",
+  hour12: true,
+  timeZone: "UTC"
+};
 
 export default function EventDate({ date, includeDate }: { date: string; includeDate: boolean }) {
-  const formatted = new Date(date)
-    .toLocaleString("en-US", DATE_OPTIONS)
-    .replaceAll(/,\s+12:00\s+AM/giu, "");
-  // if we do not include date, split the string and return the last two words
-  // so that for all-day events we get the dates and for non-all-day events we get the time
+  const dateObj = new Date(date);
+  const isAllDay = dateObj.getUTCHours() === 0 && dateObj.getUTCMinutes() === 0;
```

Comment:
Does this mean events hosted at 0:00 UTC (8:00 PM our time) will be treated as "all day" events even if they last say. an hour?

Reply:
It actually works because we store our events as if UTC was EST time. I.e. an event at 12:00AM EST is stored as 12:00AM UTC.

Reply:
Seems to work tho. I think this is because we store things in UTC time even if they are EST.
<img width="1297" alt="image" src="https://github.com/user-attachments/assets/d7635874-a84d-4a01-b904-7960eef6b9ac">

Reply:
Ah, I see. The point still stands tho, if I host my event at midnight.

Reply:
![image](https://github.com/user-attachments/assets/71549497-55ae-4a62-bd5d-6bbfd63e29bb)

Lmk how it shows up

Reply:
Yeah, you do have a point. We had this workaround because the LLM will tend to output 12:00AM for events that are all-day or have no specified time. So we made this tradeoff to ignore events that happen at midnight